bidwire/scrapers/knox_tn_agendas_scraper.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import scrapelib
+from lxml import html
+from urllib import parse
+
+from sqlalchemy.exc import IntegrityError
+
+from document import Document
+from scrapers.base_scraper import BaseScraper
+
+log = logging.getLogger(__name__)
+
+
+class KnoxCoTNAgendaScraper(BaseScraper):
+    SITE_ROOT_URL = 'https://destinyhosted.com/'
+    MEETING_SCHEDULE_URL = 'https://destinyhosted.com/agenda_publish.cfm?id=56691&mt=ALL'
+
+    def __init__(self) -> None:
+        self.scraper = scrapelib.Scraper()
+
+    def scrape(self, session):
+        page = self.scraper.get(self.MEETING_SCHEDULE_URL)
+        documents = self._get_docs_from_schedule(page.content)
+        log.debug("Found %d documents", len(documents))
+        new_docs = []
+        for doc in documents:
+            try:
+                with session.begin_nested():
+                    session.add(doc)
+            except IntegrityError:
```

Comment:
It is going to be a separate roundtrip to the DB but it's likely not much (if at all) slower than doing the query/insert method since the data set is so small.

`begin_nested()` starts a savepoint in PostgreSQL so that you can work in smaller chunks within a larger transaction.